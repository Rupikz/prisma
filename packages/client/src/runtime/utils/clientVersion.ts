// @ts-ignore because it's out of src and typescript will complain
export { version as clientVersion } from '../../../package.json'
