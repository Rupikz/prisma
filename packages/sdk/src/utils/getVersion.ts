// @ts-ignore because it's out of src and typescript will complain
export { version } from '../../package.json'
