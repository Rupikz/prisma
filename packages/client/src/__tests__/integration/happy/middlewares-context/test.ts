import { getTestClient } from '../../../../utils/getTestClient'

describe('context', () => {
  test('basic', async () => {
    const PrismaClient = await getTestClient()

    const db = new PrismaClient()

    db.$use((params, next) => {
      expect(params).toMatchSnapshot()
      return next(params)
    })

    await db.user.findMany()

    await db.$disconnect()
  })

  test('findMany', async () => {
    const PrismaClient = await getTestClient()

    const db = new PrismaClient()

    db.$use((params, next) => {
      expect(params).toMatchSnapshot()
      return next(params)
    })

    await db.user.findMany({
      where: {
        id: {
          in: ['1', '2'],
        },
      },
      context: {
        test: 'test',
      },
    })

    await db.$disconnect()
  })

  test('findFirst', async () => {
    const PrismaClient = await getTestClient()

    const db = new PrismaClient()

    db.$use((params, next) => {
      expect(params).toMatchSnapshot()
      return next(params)
    })

    await db.user.findFirst({
      select: {
        id: true,
        name: true,
      },
      where: {
        name: 'test',
        email: 'test@example.com',
      },
      context: {
        test: {
          test: {
            test: 'test',
          },
        },
      },
    })

    await db.$disconnect()
  })
})
