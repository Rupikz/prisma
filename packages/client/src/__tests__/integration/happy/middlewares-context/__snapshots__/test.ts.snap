// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`context basic 1`] = `
Object {
  action: findMany,
  args: undefined,
  context: undefined,
  dataPath: Array [],
  model: User,
  runInTransaction: false,
}
`;

exports[`context findMany 1`] = `
Object {
  action: findMany,
  args: Object {
    where: Object {
      id: Object {
        in: Array [
          1,
          2,
        ],
      },
    },
  },
  context: Object {
    test: test,
  },
  dataPath: Array [],
  model: User,
  runInTransaction: false,
}
`;

exports[`context findFirst 1`] = `
Object {
  action: findFirst,
  args: Object {
    select: Object {
      id: true,
      name: true,
    },
    where: Object {
      email: test@example.com,
      name: test,
    },
  },
  context: Object {
    test: Object {
      test: Object {
        test: test,
      },
    },
  },
  dataPath: Array [],
  model: User,
  runInTransaction: false,
}
`;
